pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -117,46 +165,135 @@ func haveOverlap(a1, a2 []string) bool {
 	return false
 }
 
+// PreFilter computes and stores cycleState containing details for enforcing ReadWriteOncePod.
 func (pl *VolumeRestrictions) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
-	if pl.enableReadWriteOncePod {
-		return nil, pl.isReadWriteOncePodAccessModeConflict(ctx, pod)
+	if !pl.enableReadWriteOncePod {
+		return nil, nil
+	}
+
+	pvcs, err := pl.readWriteOncePodPVCsForPod(ctx, pod)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, framework.NewStatus(framework.UnschedulableAndUnresolvable, err.Error())
+		}
+		return nil, framework.AsStatus(err)
+	}
+
+	s, err := pl.calPreFilterState(ctx, pod, pvcs)
+	if err != nil {
+		return nil, framework.AsStatus(err)
+	}
+	cycleState.Write(preFilterStateKey, s)
+	return nil, nil
```

Comment:
You could return `Unschedulable` in `PreFilter` in some scenarios to completely skip Filter, right?

Probably:

```golang
return nil, satisfyReadWriteOncePod(ctx, s)
```

Reply:
Having trouble making this work:
```
=== RUN   TestReadWriteOncePodPreemption/preempt_single_pod
E1212 14:10:24.194140 1682349 framework.go:698] "Failed running RemovePod on PreFilter plugin" err="reading \"PreFilterPodTopologySpread\" from cycleState: not found" plugin="PodTopologySpread" pod="preemptionc64798a8-c9ef-4911-9a89-885ea274900e/preemptor-pod"
E1212 14:10:24.194204 1682349 schedule_one.go:146] "Status after running PostFilter plugins for pod" pod="preemptionc64798a8-c9ef-4911-9a89-885ea274900e/preemptor-pod" status=&{code:1 reasons:[preemption: running RemovePod on PreFilter plugin "PodTopologySpread": reading "PreFilterPodTopologySpread" from cycleState: not found] err:0xc005bfe4f0 failedPlugin:}
    preemption_test.go:1940: Pod preemptionc64798a8-c9ef-4911-9a89-885ea274900e/victim-pod is not getting evicted.
```

Reply:
you still have to write into the `cycleState`. Did you try that?

Reply:
Synced offline -- will leave this as a followup, see #114433